### ISSUE TITLE
no calendar entries in next month

### ISSUE THREAD

**Issue by Hoppaz** [nextcloud/calendar#637]
### Steps to reproduce
1. in monthly view click on next month - month appears empty
2. then click on "today" (heute) and even the current month appears empty
3. reloading the page shows again the entries of the current month

If I click on week after 2. the week entries of the current week are shown. After that clicking on monthly view shows the entries of the month again.

firefox debug console shows the following every time the problem appears:
Object { error: true, redirect: false, message: "Der entfernte Server erlaubt keinen…" }

I expect that this problem comes with the latest update.

**Comment by georgehrke** [comment]
@Hoppaz Can you please post the entire javascript log? 
The little german snippet doesn't say a lot ...

**Comment by georgehrke** [comment]
can't reproduce

**Comment by Hoppaz** [comment]
argh... my fault!
... of using proxy settings in config.php

Thank you for your work and sorry for wasting your time...

**Comment by Hoppaz** [comment]
Maybe a true problem with proxy setting.

If I set the proxy in config.php to my proxy to access the internet the problem appears.

It seems that if application cannot reach the internet via proxy calendar works (proxy not set in config.php).

If I set it ( proxy => 'myproxy:3129' )  I can see that calendar strips the domain from an remote calendar ( using a calendar from www.webcal.fi ) while trying to retrieve the remove content within the squid log and the problem appears.

**Comment by Hoppaz** [comment]
-> https://help.nextcloud.com/t/js-error-when-synchronising-external-calendar/20320

Seems to be a general problem with external calendar.

As I wrote in the last comment it looks like the host gets stripped from the URL.

**Comment by georgehrke** [comment]
@Hoppaz Can you use the issue template and provide logs anyway please? :) Thx!

**Comment by Hoppaz** [comment]
The behaviour changed a little.

I have removed the web calendar. 

If I try to add http://www.webcal.fi/cal.php?id=75&format=ics&wrn=1&wp=4&wf=53&color=%23FF3100&cntr=de&lang=de&rid=wc
I get an error message "Der entfernte Server erlaubt keinen Zugriff auf den Kalender (HTTP {400} Fehler)".


### General server configuration

**Operating system:** Linux nextcloud 4.9.0-4-amd64 #1 SMP Debian 4.9.51-1 (2017-09-28) x86_64

**Web server:** nginx/1.10.3 (fpm-fcgi)

**Database:** mysql 10.1.26

**PHP version:** 7.0.19-1
<details>
        <summary>PHP-modules loaded</summary>

```
 - Core
 - date
 - libxml
 - openssl
 - pcre
 - zlib
 - filter
 - hash
 - Reflection
 - SPL
 - session
 - standard
 - cgi-fcgi
 - mysqlnd
 - PDO
 - xml
 - apcu
 - apc
 - calendar
 - ctype
 - curl
 - dom
 - mbstring
 - fileinfo
 - ftp
 - gd
 - gettext
 - iconv
 - igbinary
 - intl
 - json
 - ldap
 - exif
 - mysqli
 - pdo_mysql
 - Phar
 - posix
 - readline
 - redis
 - shmop
 - SimpleXML
 - sockets
 - sysvmsg
 - sysvsem
 - sysvshm
 - tokenizer
 - wddx
 - xmlreader
 - xmlwriter
 - xsl
 - zip
 - Zend OPcache
```
</details>

### Nextcloud configuration

**Nextcloud version:** 12.0.3 - 12.0.3.3

**Updated from an older Nextcloud/ownCloud or fresh install:** Fresh install

**Where did you install Nextcloud from:**  nextcloud.com


**Are you using external storage, if yes which one:** files_external is disabled

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

<details>
        <summary>Signing status</summary>

```
[]
```
</details>

<details>
        <summary>Enabled apps</summary>

```
 - activity: 2.5.2
 - admin_audit: 1.2.0
 - admin_notifications: 1.0.0
 - bruteforcesettings: 1.0.2
 - calendar: 1.5.6
 - checksum: 0.3.5
 - circles: 0.13.3
 - cms_pico: 0.9.6
 - comments: 1.2.0
 - contacts: 2.0.1
 - dashboard: 4.0.5
 - dav: 1.3.0
 - external: 2.0.3
 - federatedfilesharing: 1.2.0
 - federation: 1.2.0
 - files: 1.7.2
 - files_clipboard: 0.6.4
 - files_downloadactivity: 1.1.1
 - files_pdfviewer: 1.1.1
 - files_retention: 1.1.2
 - files_sharing: 1.4.0
 - files_texteditor: 2.4.1
 - files_trashbin: 1.2.0
 - files_versions: 1.5.0
 - files_videoplayer: 1.1.0
 - firstrunwizard: 2.1
 - gallery: 17.0.0
 - issuetemplate: 0.2.2
 - logreader: 2.0.0
 - lookup_server_connector: 1.0.0
 - mail: 0.7.3
 - metadata: 0.5.0
 - nextcloud_announcements: 1.1
 - notes: 2.3.1
 - notifications: 2.0.0
 - oauth2: 1.0.5
 - ocsms: 1.12.1
 - onlyoffice: 1.1.5
 - password_policy: 1.2.2
 - phonetrack: 0.0.5
 - polls: 0.8.0
 - previewgenerator: 1.0.7
 - provisioning_api: 1.2.0
 - serverinfo: 1.2.0
 - sharebymail: 1.2.0
 - socialsharing_email: 1.0.1
 - spreed: 2.0.1
 - spreedme: 0.3.11
 - survey_client: 1.0.0
 - systemtags: 1.2.0
 - tasks: 0.9.5
 - theming: 1.3.0
 - twofactor_backupcodes: 1.1.1
 - twofactor_totp: 1.3.1
 - updatenotification: 1.2.0
 - user_external: 0.4
 - workflowengine: 1.2.0
```
</details>

<details>
        <summary>Disabled apps</summary>

```
 - drawio
 - encryption
 - files_external
 - ownpad
 - rainloop
 - user_ldap
 - weather
 - workin2gether
```
</details>

<details>
        <summary>Content of config/config.php</summary>

```
{
    "instanceid": "ocfc54378fch",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "10.1.3.20",
        "nc.peamx.de"
    ],
    "proxy": "10.1.3.1:3128",
    "memcache.local": "\\OC\\Memcache\\Redis",
    "memcache.distributed": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "localhost",
        "port": 6379
    },
    "datadirectory": "\/var\/www\/data",
    "overwrite.cli.url": "http:\/\/10.1.3.20",
    "dbtype": "mysql",
    "version": "12.0.3.3",
    "dbname": "nextcloud",
    "dbhost": "10.1.1.105",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "trusted_proxies": [
        "10.1.3.10"
    ],
    "forwarded_for_headers": [
        "HTTP_X_FORWARDED_FOR"
    ],
    "mail_from_address": "foobar",
    "mail_smtpmode": "php",
    "mail_smtpauthtype": "LOGIN",
    "mail_domain": "peamx.de",
    "maintenance": false,
    "theme": "",
    "loglevel": 2
}
```
</details>


### Client configuration

**Browser:** Mozilla/5.0 (X11; Linux x86_64; rv:52.0) Gecko/20100101 Firefox/52.0

**Operating system:** debian 9

### Logs

<details>
        <summary>Nextcloud log (data/nextcloud.log)</summary>

```
No new entries on adding the web calendar.
```
</details>

<details>
<summary>Proxy log</summary>

I can see the following entry in the squid log:
1508507931.894      0 10.1.3.20 TAG_NONE/400 4080 GET /cal.php?id=75&format=ics&wrn=1&wp=4&wf=53&color=%23FF3100&cntr=de&lang=de&rid=wc - HIER_NONE/- text/html

If I do it with wget on the command line of the nextcloud server it works:

root@nextcloud:/var/www/data# export http_proxy=10.1.3.1:3128
root@nextcloud:/var/www/data# LANG=C wget "http://www.webcal.fi/cal.php?id=75&format=ics&wrn=1&wp=4&wf=53&color=%23FF3100&cntr=de&lang=de&rid=wc" -O cal.ics
--2017-10-20 16:03:29--  http://www.webcal.fi/cal.php?id=75&format=ics&wrn=1&wp=4&wf=53&color=%23FF3100&cntr=de&lang=de&rid=wc
Connecting to 10.1.3.1:3128... connected.
Proxy request sent, awaiting response... 200 OK
Length: unspecified [text/calendar]
Saving to: 'cal.ics'

cal.ics                                                  [ <=>                                                                                                                  ]  32.71K  --.-KB/s    in 0.003s  

2017-10-20 16:03:29 (12.3 MB/s) - 'cal.ics' saved [33492]

and the corresponding entry in squid log looks like this:
1508508190.879     52 10.1.3.20 TCP_MISS/200 34559 GET http://www.webcal.fi/cal.php? - HIER_DIRECT/104.28.31.21 text/calendar

</details>

**Comment by tcitworld** [comment]
Logs in debug mode:
```
[Fri Oct 20 16:39:36 2017] Exception: {"Exception":"InvalidArgumentException","Message":"Unable to parse malformed url: https:\/\/\/remote.php\/dav\/www.webcal.fi\/cal.php?id=75&format=ics&wrn=1&wp=4&wf=53&color=%23FF3100&cntr=de&lang=de&rid=wc","Code":0,"Trace":"#0 \/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Message\/Request.php(70): GuzzleHttp\\Url::fromString('https:\/\/\/remote...')\n#1 \/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Message\/Request.php(42): GuzzleHttp\\Message\\Request->setUrl('https:\/\/\/remote...')\n#2 \/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Message\/MessageFactory.php(85): GuzzleHttp\\Message\\Request->__construct('GET', 'https:\/\/\/remote...', Array, NULL, Array)\n#3 \/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Client.php(120): GuzzleHttp\\Message\\MessageFactory->createRequest('GET', 'https:\/\/\/remote...', Array)\n#4 \/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Client.php(125): GuzzleHttp\\Client->createRequest('GET', 'https:\/\/\/remote...', Array)\n#5 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/Http\/Client\/Client.php(138): GuzzleHttp\\Client->get('https:\/\/\/remote...', Array)\n#6 \/home\/tcit\/dev\/nextcloud\/server\/apps\/calendar\/controller\/proxycontroller.php(90): OC\\Http\\Client\\Client->get('https:\/\/\/remote...', Array)\n#7 [internal function]: OCA\\Calendar\\Controller\\ProxyController->proxy('https:\/\/\/remote...')\n#8 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/AppFramework\/Http\/Dispatcher.php(160): call_user_func_array(Array, Array)\n#9 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/AppFramework\/Http\/Dispatcher.php(90): OC\\AppFramework\\Http\\Dispatcher->executeController(Object(OCA\\Calendar\\Controller\\ProxyController), 'proxy')\n#10 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/AppFramework\/App.php(114): OC\\AppFramework\\Http\\Dispatcher->dispatch(Object(OCA\\Calendar\\Controller\\ProxyController), 'proxy')\n#11 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php(47): OC\\AppFramework\\App::main('ProxyController', 'proxy', Object(OC\\AppFramework\\DependencyInjection\\DIContainer), Array)\n#12 [internal function]: OC\\AppFramework\\Routing\\RouteActionHandler->__invoke(Array)\n#13 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/Route\/Router.php(297): call_user_func(Object(OC\\AppFramework\\Routing\\RouteActionHandler), Array)\n#14 \/home\/tcit\/dev\/nextcloud\/server\/lib\/base.php(985): OC\\Route\\Router->match('\/apps\/calendar\/...')\n#15 \/home\/tcit\/dev\/nextcloud\/server\/index.php(40): OC::handleRequest()\n#16 {main}","File":"\/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Url.php","Line":39}
```
```
[Fri Oct 20 16:39:37 2017] Exception: {"Exception":"InvalidArgumentException","Message":"Unable to parse malformed url: http:\/\/\/remote.php\/dav\/www.webcal.fi\/cal.php?id=75&format=ics&wrn=1&wp=4&wf=53&color=%23FF3100&cntr=de&lang=de&rid=wc","Code":0,"Trace":"#0 \/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Message\/Request.php(70): GuzzleHttp\\Url::fromString('http:\/\/\/remote....')\n#1 \/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Message\/Request.php(42): GuzzleHttp\\Message\\Request->setUrl('http:\/\/\/remote....')\n#2 \/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Message\/MessageFactory.php(85): GuzzleHttp\\Message\\Request->__construct('GET', 'http:\/\/\/remote....', Array, NULL, Array)\n#3 \/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Client.php(120): GuzzleHttp\\Message\\MessageFactory->createRequest('GET', 'http:\/\/\/remote....', Array)\n#4 \/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Client.php(125): GuzzleHttp\\Client->createRequest('GET', 'http:\/\/\/remote....', Array)\n#5 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/Http\/Client\/Client.php(138): GuzzleHttp\\Client->get('http:\/\/\/remote....', Array)\n#6 \/home\/tcit\/dev\/nextcloud\/server\/apps\/calendar\/controller\/proxycontroller.php(90): OC\\Http\\Client\\Client->get('http:\/\/\/remote....', Array)\n#7 [internal function]: OCA\\Calendar\\Controller\\ProxyController->proxy('http:\/\/\/remote....')\n#8 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/AppFramework\/Http\/Dispatcher.php(160): call_user_func_array(Array, Array)\n#9 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/AppFramework\/Http\/Dispatcher.php(90): OC\\AppFramework\\Http\\Dispatcher->executeController(Object(OCA\\Calendar\\Controller\\ProxyController), 'proxy')\n#10 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/AppFramework\/App.php(114): OC\\AppFramework\\Http\\Dispatcher->dispatch(Object(OCA\\Calendar\\Controller\\ProxyController), 'proxy')\n#11 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php(47): OC\\AppFramework\\App::main('ProxyController', 'proxy', Object(OC\\AppFramework\\DependencyInjection\\DIContainer), Array)\n#12 [internal function]: OC\\AppFramework\\Routing\\RouteActionHandler->__invoke(Array)\n#13 \/home\/tcit\/dev\/nextcloud\/server\/lib\/private\/Route\/Router.php(297): call_user_func(Object(OC\\AppFramework\\Routing\\RouteActionHandler), Array)\n#14 \/home\/tcit\/dev\/nextcloud\/server\/lib\/base.php(985): OC\\Route\\Router->match('\/apps\/calendar\/...')\n#15 \/home\/tcit\/dev\/nextcloud\/server\/index.php(40): OC::handleRequest()\n#16 {main}","File":"\/home\/tcit\/dev\/nextcloud\/server\/3rdparty\/guzzlehttp\/guzzle\/src\/Url.php","Line":39}
```

**Comment by sbrl** [comment]
I can independently confirm this bug on my installation. I've discovered it happens when you try and navigate when the calendars and the subscriptions haven't finished loading yet.

**Comment by georgehrke** [comment]
>  I've discovered it happens when you try and navigate when the calendars and the subscriptions haven't finished loading yet.

Very important detail! Thx for the hint :)

**Comment by ou-li** [comment]
I can confirm this. Having an external calendar
(http://www.ifeiertage.de/sl-.ics - server down at the moment) that fails loading makes next months appear empty.

**Comment by georgehrke** [comment]
As far as I can tell this will be fixed with https://github.com/nextcloud/calendar/pull/707/files#diff-ed42dc2bca6b9081ba3850baa1a59ebbR117

This patch will be released with 1.6.0 in the next couple of days. Please speak up if it was not fixed.
